jni/com_eclipsesource_v8_V8Impl.cpp

```diff
@@ -1686,23 +1691,32 @@ JNIEXPORT jlong JNICALL Java_com_eclipsesource_v8_V8__1registerJavaMethod
   isolate->IdleNotificationDeadline(1);
   MethodDescriptor* md= new MethodDescriptor();
   Local<External> ext = External::New(isolate, md);
-  Persistent<External> pext(isolate, ext);
-  pext.SetWeak(md, [](v8::WeakCallbackInfo<MethodDescriptor> const& data) {
-    MethodDescriptor* md = data.GetParameter();
-    jobject v8 = reinterpret_cast<V8Runtime*>(md->v8RuntimePtr)->v8;
-    JNIEnv * env;
-    getJNIEnv(env);
-    env->CallVoidMethod(v8, v8DisposeMethodID, md->methodID);
-    delete(md);
-  }, WeakCallbackType::kParameter);
 
   md->methodID = reinterpret_cast<jlong>(md);
   md->v8RuntimePtr = v8RuntimePtr;
+
   MaybeLocal<Function> func = Function::New(context, callback, ext);
   if (!func.IsEmpty()) {
     Maybe<bool> unusedResult = object->Set(context, v8FunctionName, func.ToLocalChecked());
     unusedResult.Check();
   }
+
+  md->obj.Reset(isolate, ext);
```

Comment:
It's duplicate code, can we avoid the duplication?

Reply:
Code duplication issue was addressed.

Is there anything missing?

Reply:
sorry for the late response. no, it looks good to me.